src/app.rs

```diff
@@ -84,11 +86,14 @@ where
         let new_html = (self.view)(&new_model);
 
         self.render_app(&new_html).unwrap();
+
+        drop(state);
```

Comment:
Since `state` is in a `Ref`, it's possible to violate the borrow checker rules. We must drop this reference or it'll crash on `set_state` below (because that requires a mutable reference).